src/proposer/proposer_rpc.cpp

```diff
@@ -167,6 +170,125 @@ class ProposerRPCImpl : public ProposerRPC {
     obj.pushKV("stakeable_coins", arr);
     return obj;
   }
+
+  UniValue ProposeBlocks(CWallet &wallet,
+                         const boost::optional<CScript> &coinbase_script,
+                         const int num_generate) const {
+
+    proposer::Proposer *proposer = GetComponent<proposer::Proposer>();
```

Comment:
I see. `GetComponent` must not be used in a component. Inject it. The proposer rpc is an injected componend.